script/utils/CurrentConfigCheck.s.sol

```diff
@@ -0,0 +1,570 @@
+        require(executorMultisigOwners.length == 2,
+            "executorMultisig owners wrong length");
+        bool timelockInOwners;
+        bool communityMultisigInOwners;
+        for (uint256 i = 0; i < 2; ++i) {
+            if (executorMultisigOwners[i] == timelock) {
+                timelockInOwners = true;
+            }
+            if (executorMultisigOwners[i] == communityMultisig) {
+                communityMultisigInOwners = true;
+            }
+        }
+        require(timelockInOwners, "timelock not in executorMultisig owners");
+        require(communityMultisigInOwners, "communityMultisig not in executorMultisig owners");
+
+        require(eigenTokenProxyAdmin != beigenTokenProxyAdmin,
+            "tokens must have different proxy admins to allow different timelock controllers");
+        require(eigenTokenTimelockController != beigenTokenTimelockController,
+            "tokens must have different timelock controllers");
+
+        // note that proxy admin owners are different but _token_ owners are the same
+        assertEq(Ownable(address(EIGEN)).owner(), address(eigenTokenTimelockController),
+            "EIGEN.owner() != eigenTokenTimelockController");
+        assertEq(Ownable(address(bEIGEN)).owner(), address(eigenTokenTimelockController),
+            "bEIGEN.owner() != eigenTokenTimelockController");
+        assertEq(eigenTokenProxyAdmin.owner(), address(eigenTokenTimelockController),
+            "eigenTokenProxyAdmin.owner() != eigenTokenTimelockController");
+        assertEq(beigenTokenProxyAdmin.owner(), address(beigenTokenTimelockController),
+            "beigenTokenProxyAdmin.owner() != beigenTokenTimelockController");
+
+        assertEq(eigenTokenProxyAdmin.getProxyAdmin(TransparentUpgradeableProxy(payable(address(EIGEN)))),
+            address(eigenTokenProxyAdmin),
+            "eigenTokenProxyAdmin is not actually the admin of the EIGEN token");
+        assertEq(beigenTokenProxyAdmin.getProxyAdmin(TransparentUpgradeableProxy(payable(address(bEIGEN)))),
+            address(beigenTokenProxyAdmin),
+            "beigenTokenProxyAdmin is not actually the admin of the bEIGEN token");
+
+        require(eigenTokenTimelockController.hasRole(eigenTokenTimelockController.PROPOSER_ROLE(), foundationMultisig),
+            "foundationMultisig does not have PROPOSER_ROLE on eigenTokenTimelockController");
+        require(eigenTokenTimelockController.hasRole(eigenTokenTimelockController.EXECUTOR_ROLE(), foundationMultisig),
+            "foundationMultisig does not have EXECUTOR_ROLE on eigenTokenTimelockController");
+        require(eigenTokenTimelockController.hasRole(eigenTokenTimelockController.CANCELLER_ROLE(), operationsMultisig),
+            "operationsMultisig does not have CANCELLER_ROLE on eigenTokenTimelockController");
+        require(eigenTokenTimelockController.hasRole(eigenTokenTimelockController.TIMELOCK_ADMIN_ROLE(), executorMultisig),
```

Comment:
we should check that the TC is its own admin

Reply:
this check is already in the `checkTimelockControllerConfig` function:
https://github.com/Layr-Labs/eigenlayer-contracts/blob/ab8f84ba5aa2fd889f50ae874e7eff4e51ee364d/script/utils/CurrentConfigCheck.s.sol#L292-L294